crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -2007,6 +2126,37 @@ async fn update_payment_method_status_and_ntid<F: Clone>(
     Ok(())
 }
 
+fn update_connector_mandate_details_for_the_flow<F: Clone>(
+    connector_mandate_id: Option<String>,
+    mandate_metadata: Option<serde_json::Value>,
+    payment_data: &mut PaymentData<F>,
+) -> RouterResult<()> {
+    let connector_mandate_reference_id = Some(ConnectorMandateReferenceId {
+        connector_mandate_id: connector_mandate_id.clone(),
```

Comment:
what if we already have some `connector_mandate_id`, what about existing or inflight payments. This would essentially override the existing data in connector mandate details right?

Reply:
No overriding won't happen as the update to the payment_method table is being handled only once